app/services/search/location_builder.rb

```diff
@@ -20,29 +20,39 @@ def initialize(location, radius, location_category, buffer_radius)
                          end
 
     if location_category_search?
-      initialize_location_polygon
+      initialize_search_polygon_boundaries
     elsif @location.present?
       @location_filter = build_location_filter(@location, @radius)
     end
   end
 
   def location_category_search?
-    (@location_category && LocationCategory.include?(@location_category)) ||
-      (@location && LocationCategory.include?(@location))
+    (location_category && LocationCategory.include?(location_category)) ||
+      (location && LocationCategory.include?(location))
   end
 
 private
 
-  def initialize_location_polygon
-    @location_polygon = LocationPolygon.with_name(@location_category)
-    if @location_polygon.present?
-      @search_polygon_boundary = if @buffer_radius.present?
-                                   [@location_polygon.buffers[@buffer_radius]]
-                                 else
-                                   [@location_polygon.boundary]
-                                 end
-    end
-    if location_polygon.nil? && (DOWNCASE_REGIONS + DOWNCASE_COUNTIES).include?(@location_category.downcase)
+  def initialize_search_polygon_boundaries
+    location_polygon = LocationPolygon.with_name(location_category)
+    if location_polygon.present?
+      @search_polygon_boundaries = if buffer_radius.present?
+                                     [location_polygon.buffers[buffer_radius]]
+                                   else
+                                     [location_polygon.boundary]
+                                   end
+    elsif DOWNCASE_COMPOSITE_LOCATIONS.include?(location_category.downcase)
```

Comment:
I'll admit being a bit bamboozled by all these different constants that mean slightly different things, and having to try and figure out what means what and why we're using which bit where. Not sure how to overcome that though